### PR TITLE
🌐 [i18n-KO] Translated `language-modeling.mdx`

### PR DESCRIPTION
**Pull request by wonhyeongseo**
# What does this PR do?

Translated the `language-modeling.mdx` file of the documentation to Korean.
Thank you in advance for your review. 

Part of https://github.com/huggingface/transformers/issues/20179

## Before reviewing
- [x] Check for missing / redundant translations (번역 누락/중복 검사)
- [x] Grammar Check (맞춤법 검사)
- [x] Review or Add new terms to glossary (용어 확인 및 추가)
- [x] Check Inline TOC (e.g. `[[lowercased-header]]`)
- [x] Check live-preview for gotchas (live-preview로 정상작동 확인)

## Who can review? (Initial)

Team PseudoLab, may you please review this PR? @0525hhgus, @KIHOON71, @sim-so, @gabrielwithappy, @HanNayeoniee, @wonhyeongseo, @jungnerd

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review? (Final)

@sgugger, @ArthurZucker, @eunseojo May you please review this PR?

### CODE REVIEW

File: docs/source/ko/tasks/language_modeling.mdx [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+
+다음 단계는 `text` 필드를 전처리하기 위해 DistilGPT2 토크나이저를 불러오는 것입니다.
+
+```py
+>>> from transformers import AutoTokenizer
+
+>>> tokenizer = AutoTokenizer.from_pretrained("distilgpt2")
+```
+
+위의 예제에서 알 수 있듯이, `text` 필드는 `answers` 아래에 중첩되어 있습니다. 따라서 [`flatten`](https://huggingface.co/docs/datasets/process.html#flatten) 메서드를 사용하여 중첩 구조에서 `text` 하위 필드를 추출해야 합니다.
```
Comment (sim-so) [inline docs/source/ko/tasks/language_modeling.mdx:106]:
```suggestion
위의 예제에서 알 수 있듯이, `text` 필드는 `answers` 아래에 중첩되어 있습니다. 따라서 [`flatten`](https://huggingface.co/docs/datasets/process.html#flatten) 메소드를 사용하여 중첩 구조에서 `text` 하위 필드를 추출해야 합니다.
```

File: docs/source/ko/tasks/language_modeling.mdx [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+각 하위 필드는 이제 `answers` 접두사를 가진 별도의 열로 나뉘었으며, `text` 필드는 이제 리스트입니다. 각 문장을 개별적으로 토큰화하는 대신, 먼저 리스트를 문자열로 변환하여 한꺼번에 토큰화할 수 있습니다.
+
+다음은 문자열 리스트를 결합하고 결과를 토큰화하는 첫 번째 전처리 함수입니다:
+
+```py
+>>> def preprocess_function(examples):
+...     return tokenizer([" ".join(x) for x in examples["answers.text"]])
+```
+
+이 전처리 함수를 전체 데이터셋에 적용하려면 🤗 Datasets [`~datasets.Dataset.map`] 메서드를 사용하세요. `batched=True`로 설정하여 데이터셋의 여러 요소를 한 번에 처리하고, `num_proc`를 증가시켜 프로세스 수를 늘릴 수 있습니다. 필요 없는 열은 제거하세요:
```
Comment (sim-so) [inline docs/source/ko/tasks/language_modeling.mdx:134]:
```suggestion
이 전처리 함수를 전체 데이터 세트에 적용하려면 🤗 Datasets [`~datasets.Dataset.map`] 메소드를 사용하세요. `batched=True`로 설정하여 데이터셋의 여러 요소를 한 번에 처리하고, `num_proc`를 증가시켜 프로세스 수를 늘릴 수 있습니다. 필요 없는 열은 제거하세요:
```

File: docs/source/ko/tasks/language_modeling.mdx [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+...     # Split by chunks of block_size.
+...     result = {
+...         k: [t[i : i + block_size] for i in range(0, total_length, block_size)]
+...         for k, t in concatenated_examples.items()
+...     }
+...     result["labels"] = result["input_ids"].copy()
+...     return result
+```
+
+전체 데이터셋에 `group_texts` 함수를 적용하세요:
```
Comment (sim-so) [inline docs/source/ko/tasks/language_modeling.mdx:172]:
```suggestion
전체 데이터 세트에 `group_texts` 함수를 적용하세요:
```

File: docs/source/ko/tasks/language_modeling.mdx [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+```py
+>>> tokenized_eli5 = eli5.map(
+...     preprocess_function,
+...     batched=True,
+...     num_proc=4,
+...     remove_columns=eli5["train"].column_names,
+... )
+```
+
+이제 데이터셋은 시퀀스가 토큰화됐지만, 일부 시퀀스는 모델의 최대 입력 길이보다 길 수 있습니다.
```
Comment (sim-so) [inline docs/source/ko/tasks/language_modeling.mdx:145]:
```suggestion
이제 데이터 세트는 시퀀스가 토큰화됐지만, 일부 시퀀스는 모델의 최대 입력 길이보다 길 수 있습니다.
```

File: docs/source/ko/tasks/language_modeling.mdx [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+
+커뮤니티에 모델을 업로드하고 공유하기 위해 Hugging Face 계정에 로그인하는 것을 권장합니다. 알림이 표시되면 토큰을 입력하여 로그인하세요:
+
+```py
+>>> from huggingface_hub import notebook_login
+
+>>> notebook_login()
+```
+
+## ELI5 데이터셋 불러오기[[load-eli5-dataset]]
```
Comment (sim-so) [inline docs/source/ko/tasks/language_modeling.mdx:57]:
```suggestion
## ELI5 데이터 세트 불러오기[[load-eli5-dataset]]
```

File: docs/source/ko/tasks/language_modeling.mdx [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+
+먼저, 🤗 Datasets 라이브러리에서 r/askscience의 작은 하위 집합인 ELI5 데이터셋을 불러옵니다.
```
Comment (sim-so) [inline docs/source/ko/tasks/language_modeling.mdx:59]:
```suggestion
먼저, 🤗 Datasets 라이브러리에서 r/askscience의 작은 하위 집합인 ELI5 데이터 세트를 불러옵니다.
```

File: docs/source/ko/tasks/language_modeling.mdx [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+이를 통해 전체 데이터셋에서 학습하는 데 더 많은 시간을 투자하기 전에, 실험해봄으로써 모든 것이 작동하는지 확인할 수 있습니다.
```
Comment (sim-so) [inline docs/source/ko/tasks/language_modeling.mdx:60]:
```suggestion
이를 통해 전체 데이터 세트에서 학습하는 데 더 많은 시간을 투자하기 전에, 실험해봄으로써 모든 것이 작동하는지 확인할 수 있습니다.
```

File: docs/source/ko/tasks/language_modeling.mdx [hunk 8]
```diff
@@ -0,0 +1,413 @@
...
+```py
+>>> from transformers import AutoModelForCausalLM, TrainingArguments, Trainer
+
+>>> model = AutoModelForCausalLM.from_pretrained("distilgpt2")
+```
+
+여기까지 진행하면 세 단계만 남았습니다:
+
+1. [`TrainingArguments`]에서 훈련 하이퍼파라미터를 정의하세요. `output_dir`은 유일한 필수 매개변수로, 모델을 저장할 위치를 지정합니다. (먼저 Hugging Face에 로그인 필수) `push_to_hub=True`로 설정하여 이 모델을 허브에 업로드할 수 있습니다.
+2. 훈련 인수를 [`Trainer`]에 모델, 데이터셋 및 데이터 콜레이터와 함께 전달하세요.
```
Comment (sim-so) [inline docs/source/ko/tasks/language_modeling.mdx:226]:
```suggestion
2. 훈련 인수를 [`Trainer`]에 모델, 데이터 세트 및 데이터 콜레이터와 함께 전달하세요.
```

File: docs/source/ko/tasks/language_modeling.mdx [hunk 9]
```diff
@@ -0,0 +1,413 @@
...
+...     args=training_args,
+...     train_dataset=lm_dataset["train"],
+...     eval_dataset=lm_dataset["test"],
+...     data_collator=data_collator,
+... )
+
+>>> trainer.train()
+```
+
+훈련이 완료되면 [`~transformers.Trainer.evaluate`] 메서드를 사용하여 모델을 평가하고 퍼플렉서티를 얻을 수 있습니다:
```
Comment (sim-so) [inline docs/source/ko/tasks/language_modeling.mdx:249]:
```suggestion
훈련이 완료되면 [`~transformers.Trainer.evaluate`] 메소드를 사용하여 모델을 평가하고 퍼플렉서티를 얻을 수 있습니다:
```

File: docs/source/ko/tasks/language_modeling.mdx [hunk 10]
```diff
@@ -0,0 +1,413 @@
...
+
+```py
+>>> import math
+
+>>> eval_results = trainer.evaluate()
+>>> print(f"Perplexity: {math.exp(eval_results['eval_loss']):.2f}")
+Perplexity: 49.61
+```
+
+그런 다음 [`~transformers.Trainer.push_to_hub`] 메서드를 사용하여 모델을 허브에 공유하세요. 이렇게 하면 누구나 모델을 사용할 수 있습니다:
```
Comment (sim-so) [inline docs/source/ko/tasks/language_modeling.mdx:259]:
```suggestion
그런 다음 [`~transformers.Trainer.push_to_hub`] 메소드를 사용하여 모델을 허브에 공유하세요. 이렇게 하면 누구나 모델을 사용할 수 있습니다:
```

File: docs/source/ko/tasks/language_modeling.mdx [hunk 11]
```diff
@@ -0,0 +1,413 @@
...
+
+그런 다음 [`TFAutoModelForCausalLM`]를 사용하여 DistilGPT2를 불러옵니다:
+
+```py
+>>> from transformers import TFAutoModelForCausalLM
+
+>>> model = TFAutoModelForCausalLM.from_pretrained("distilgpt2")
+```
+
+[`~transformers.TFPreTrainedModel.prepare_tf_dataset`]을 사용하여 데이터셋을 `tf.data.Dataset` 형식으로 변환하세요:
```
Comment (sim-so) [inline docs/source/ko/tasks/language_modeling.mdx:287]:
```suggestion
[`~transformers.TFPreTrainedModel.prepare_tf_dataset`]을 사용하여 데이터 세트를 `tf.data.Dataset` 형식으로 변환하세요:
```

File: docs/source/ko/tasks/language_modeling.mdx [hunk 12]
```diff
@@ -0,0 +1,413 @@
...
+```py
+>>> from transformers.keras_callbacks import PushToHubCallback
+
+>>> callback = PushToHubCallback(
+...     output_dir="my_awesome_eli5_clm-model",
+...     tokenizer=tokenizer,
+... )
+```
+
+마지막으로, 모델을 훈련하기 위해 [`fit`](https://keras.io/api/models/model_training_apis/#fit-method)을 호출하세요. 훈련 데이터셋, 검증 데이터셋, epoch 수 및 콜백을 전달하세요:
```
Comment (sim-so) [inline docs/source/ko/tasks/language_modeling.mdx:324]: glossary를 참고하여 `epoch`을 `에폭`으로 수정합니다.
```suggestion
마지막으로, 모델을 훈련하기 위해 [`fit`](https://keras.io/api/models/model_training_apis/#fit-method)을 호출하세요. 훈련 데이터 세트, 검증 데이터 세트, 에폭 수 및 콜백을 전달하세요:
```

File: docs/source/ko/tasks/language_modeling.mdx [hunk 13]
```diff
@@ -0,0 +1,413 @@
...
+텍스트를 토큰화하고 `input_ids`를 PyTorch 텐서로 반환하세요:
+
+```py
+>>> from transformers import AutoTokenizer
+
+>>> tokenizer = AutoTokenizer.from_pretrained("my_awesome_eli5_clm-model")
+>>> inputs = tokenizer(prompt, return_tensors="pt").input_ids
+```
+
+[`~transformers.generation_utils.GenerationMixin.generate`] 메서드를 사용하여 텍스트를 생성하세요. 생성을 제어하는 다양한 텍스트 생성 전략과 매개변수에 대한 자세한 내용은 [텍스트 생성 전략](../generation_strategies) 페이지를 확인하세요.
```
Comment (sim-so) [inline docs/source/ko/tasks/language_modeling.mdx:371]:
```suggestion
[`~transformers.generation_utils.GenerationMixin.generate`] 메소드를 사용하여 텍스트를 생성하세요. 생성을 제어하는 다양한 텍스트 생성 전략과 매개변수에 대한 자세한 내용은 [텍스트 생성 전략](../generation_strategies) 페이지를 확인하세요.
```

File: docs/source/ko/tasks/language_modeling.mdx [hunk 14]
```diff
@@ -0,0 +1,413 @@
...
+텍스트를 토큰화하고 `input_ids`를 TensorFlow 텐서로 반환하세요:
+
+```py
+>>> from transformers import AutoTokenizer
+
+>>> tokenizer = AutoTokenizer.from_pretrained("my_awesome_eli5_clm-model")
+>>> inputs = tokenizer(prompt, return_tensors="tf").input_ids
+```
+
+[`~transformers.generation_tf_utils.TFGenerationMixin.generate`] 메서드를 사용하여 요약을 생성하세요. 생성을 제어하는 다양한 텍스트 생성 전략과 매개변수에 대한 자세한 내용은 [텍스트 생성 전략](../generation_strategies) 페이지를 확인하세요.
```
Comment (sim-so) [inline docs/source/ko/tasks/language_modeling.mdx:397]:
```suggestion
[`~transformers.generation_tf_utils.TFGenerationMixin.generate`] 메소드를 사용하여 요약을 생성하세요. 생성을 제어하는 다양한 텍스트 생성 전략과 매개변수에 대한 자세한 내용은 [텍스트 생성 전략](../generation_strategies) 페이지를 확인하세요.
```

**Review by sim-so — CHANGES_REQUESTED** [review]
한 번에 많은 문서를 번역하시느라 고생 많으셨어요!
glossary 관련 의견 몇 가지를 전달 드려요.

File: docs/source/ko/_toctree.yml [hunk 15]
```diff
@@ -30,8 +30,8 @@
         title: 토큰 분류
       - local: tasks/question_answering
         title: 질의 응답(Question Answering)
-      - local: in_translation
-        title: (번역중) Causal language modeling
+      - local: tasks/language_modeling
+        title: 인과 언어 모델링(Causal language modeling)
```
[the remark this answers is not in the document]
Comment (wonhyeongseo) [reply docs/source/ko/_toctree.yml:34]:
```suggestion
        title: 인과적 언어 모델링(Causal language modeling)
```
정확하십니다. 제가 용어집을 꼼꼼히 살피지 못했어요. 잡아주셔서 감사합니다!

**Review by wonhyeongseo — COMMENTED** [review]
@sim-so 님 감사합니다!

**Comment by wonhyeongseo** [comment]
@sgugger, @ArthurZucker, @eunseojo May you please review this PR?

**Review by ArthurZucker — APPROVED** [review]
Looks good to go! Thanks for the contribution 😉
